Tests work on f25

romanofski opened the pull request:
This now adds a few bugfixes and uses multiple Fedora versions to run the tests on. I've moved most of the code to run tests to the Makefile.

The F25 tests have one failing test, which I'm currently not sure if it's a regression in Encode or expected behvaiour (filed dankogai/p5-encode#88  for that).

Commit:
Use the make to run tests on multiple platforms

This adds a dockertest target to conveniently run all tests on test_*
docker images.

Commit:
Fix exception when called with a scalar argument

This feature was possible by an experimental feature introduced in Perl
5.14, but removed in 5.23, thus causing an exception in Fedora 24 with a
Perl version of 5.24

Bug: 1410050

Commit:
Make the file tests not rely on hardcoded strings

Instead of hardcoding the expectation which will pin us to specific file
versions, use the output of the file command to assert that it is
correctly working.

danc86 commented on dev-tools/docker/docker-compose.yml:
```diff
@@ -1,8 +1,14 @@
+    - &code  ../../:/code
 
+# stable release
+test_stable:
```
I would suggest to just call this `test_f25`, makes it clear what it actually is. (I assume you mean "stable" as in the latest release of Fedora but that's not really clear).

romanofski replied:
Yeh true.

Commit:
Adjust the input for sanitize_text testing

This adjusts the input string based on a possible regression with
Encode (see https://github.com/dankogai/p5-encode/issues/88).

danc86 commented on t/RPM/Grill/Util/10sanitize_text.t:
```diff
@@ -16,7 +16,7 @@ my @tests = (
     [ 'cccd'              => 'ccc[^V]d'            ], # ctrl character
     [ 'Muñoz'               => 'Muñoz'               ], # UTF-8 passthru
-    [ "Mu\361oz"            => 'Mu&amp;#241;oz'      ], # GAH! non-utf8
+    [ "Mu\361loloz"         => 'Mu&amp;#241;loloz'   ], # GAH! non-utf8
```
🤣
